TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -716,19 +719,48 @@ private static SymbolReference ResolveProcedureName(SymbolTable table, SymbolRef
             return null;
         }
 
-        protected static void Check<T>(T node, [NotNull] IList<T> found) where T : Node
+        protected static void Check<T>(string nodeTypeName, T node, [NotNull] IList<T> found) where T : Node
```

Comment:
Maybe we should mark this method as `private` instead of `protected` because it assumes that T is either Section or Paragraph to work properly. Inheritors may not be aware of that constraint, there are currently no inheritors so this is optional.

Reply:
done
https://github.com/TypeCobolTeam/TypeCobol/pull/1479/commits/b3bf2dca5a1639dffc0b0068fe759f303b177681